### PR TITLE
docs(rxPermission): FRMW-189-Migrate rxPermission to API Docs

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-189

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Dev LGTM

Migrate `rxPermission` to API Documentation.

### CODE REVIEW

**Comment by ssalinas24** [comment]
LGTM
However, I am still unable to select the checkboxes. Do i just not have permissions set or something?

**Comment by bry4n** [comment]
LGTM

File: src/rxPermission/README.md [hunk 1]
```diff
@@ -1,21 +1,3 @@
...
-
-## `hasRole(roles)`
-Given an array of roles (strings), this method returns `true` if the user has at least one of those roles, and `false` otherwise.
-
-## `hasAllRoles(roles)`
-Given an array of roles (strings), this method returns `true` if the user has _all_ of the roles, and `false` otherwise.
-
-# `rxPermission`
-This directive can be used to hide or show content based on whether or not the user has the specified role. See the demo below for an example.
+`rxPermission` component manages permissions in EncoreUI.
```
Comment (CITguy) [inline src/rxPermission/README.md]: This component doesn't manage permissions (update, add, etc.). It merely provides functionality to perform checks against existing permissions.

File: src/rxPermission/rxPermission.js [hunk 2]
```diff
@@ -19,6 +20,22 @@ angular.module('encore.ui.rxPermission', ['encore.ui.rxSession'])
...
+ *
+ * ## `getRoles()`
+ * This method takes no arguments, and returns all the roles tied to the user, in the exact format available in their 
+ * Session token.
+ *
+ * ## `hasRole(roles)` 
+ * Given an array of roles (strings), this method returns `true` if the user has at least **one** of those roles, and 
+ * `false` otherwise.
+ *
+ * ## `hasAllRoles(roles)`
```
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: These methods can probably be documented directly within the service via `@methodOf` tags. Please update.

**Comment by 100stacks** [comment]
@CITguy @bry4n @ssalinas24 okay changes made. :+1:

File: src/rxPermission/rxPermission.js [hunk 3]
```diff
@@ -59,8 +61,12 @@ angular.module('encore.ui.rxPermission', ['encore.ui.rxSession'])
...
 
-    /*
-     * @description Returns a list of all roles associated to the user
+    /**
+     * @name  getRoles
+     * @ngdoc method
+     * @methodOf rxPermission.service:Permission
+     * @description Returns a list of all roles associated to the user.  This method takes no arguments, 
+     * and returns all the roles tied to the user, in the exact format available in their Session token.
      */
```
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: * add `@returns` tag
* update description accordingly

```
/**
 * @name getRoles
 * @ngdoc method
 * @methodOf rxPermission.service:Permission
 * @description 
 * Fetch all the roles tied to the user (in the exact format available in their Session token).
 * 
 * @returns {Array} List of all roles associated to the user.
 */
```

File: src/rxPermission/rxPermission.js [hunk 4]
```diff
@@ -68,16 +74,23 @@ angular.module('encore.ui.rxPermission', ['encore.ui.rxSession'])
...
 
-    /*
-     * @description Returns whether or not the user has at least one of `roles`
+    /**
+     * @name  hasRoles
+     * @ngdoc method
+     * @methodOf rxPermission.service:Permission 
+     * @description Returns `true` or `false` based on whether or not the user has at least _one_ of `roles`.
      * @param {array} roles - List of roles to check against
      */
```
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: * add `@returns` tag
* remove hyphen at beginning of `@param` description
* update description accordingly

```
/**
 * @name hasRoles
 * @ngdoc method
 * @methodOf rxPermission.service:Permission
 * @description Check if user has at least _one_ of the given roles.
 * @param {Array} roles List of roles to check against
 * @returns {Boolean}
 */
```

File: src/rxPermission/rxPermission.js [hunk 5]
```diff
@@ -68,16 +74,23 @@ angular.module('encore.ui.rxPermission', ['encore.ui.rxSession'])
...
-    /*
-     * @description Returns whether or not the user has _every_ role in `roles`
+    /**
+     * @name  hasAllRoles
+     * @ngdoc method
+     * @methodOf rxPermission.service:Permission 
+     * @description Returns `true` if the user has _every_ role in `roles`; otherwise, it returns `false`.
+     * 
      * @param {array} roles - List of roles to check against
      */
```
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: * add `@returns` tag
* remove hyphen at beginning of `@param` description
* update description accordingly

```
/**
 * @name hasAllRoles
 * @ngdoc method
 * @methodOf rxPermission.service:Permission
 * @description Check if user has _every_ role in given list.
 * @param {Array} roles List of roles to check against
 * @returns {Boolean}
 */
```

**Comment by CITguy** [comment]
Latest feedback will produce more informative and concise API documentation.

File: src/rxPermission/rxPermission.js (hunk 4 above)
Comment (100stacks) [inline src/rxPermission/rxPermission.js]: @CITguy I modified `@returns {Boolean}`  with `@returns {Boolean} Returns `true` if user has at least _one_ of the given roles.` The ngdocs compiler appears to expect a `[description]`.
```
Running "ngdocs:all" (ngdocs) task
Generating Documentation...
Warning: Not a valid 'returns' format: {Boolean} (found in: src/rxPermission/rxPermission.js:91) Use --force to continue.

Aborted due to warnings.
Completed in 31.131s at Mon Oct 05 2015 16:52:22 GMT-0500 (CDT) 
```

File: src/rxPermission/rxPermission.js (hunk 3 above)
Comment (100stacks) [inline src/rxPermission/rxPermission.js]: @CITguy okay :+1:

File: src/rxPermission/rxPermission.js (hunk 5 above)
Comment (100stacks) [inline src/rxPermission/rxPermission.js]: @CITguy doing same here.  Please let me know if you prefer another statement on the return.
Change `@returns {Boolean}`  with `@returns {Boolean} Returns `true` if user has _every_ role in given list.`

File: src/rxPermission/rxPermission.js (hunk 4 above)
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: According to [jsdoc](http://usejsdoc.org/tags-returns.html), the `@returns` tag takes an optional description. I just tried it, but it seems that **ngdocs** expects a description for the `@returns` tag (facepalm). 

Proceed with descriptions included in the `@returns` tags.

File: src/rxPermission/rxPermission.js (hunk 5 above)
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: previous comment updated

File: src/rxPermission/rxPermission.js (hunk 4 above)
Comment (CITguy) [inline src/rxPermission/rxPermission.js]: initial comment updated

Comment (100stacks) [inline src/rxPermission/rxPermission.js]: @CITguy thanks for the jsdoc reference.  It seems better documented than __ngdocs.__

**Comment by 100stacks** [comment]
@CITguy comments are incorporated. :ok_hand:

**Comment by CITguy** [comment]
LGTM, nice work

**Pull request merged**